### DISCUSSION TITLE
Installation Issue

### DISCUSSION THREAD

**Discussion by neilpdj** [artisan-roaster-scope/artisan#1241] [Q&A] [answered]
I’m trying to install the latest version on a Windows 10 machine. The hardware is older but it’s a fresh install of windows. 
installation of Artisan succeeds but then an error occurs when I try to run it. The error states it cannot find the PyQt5 module. When I look in the installation folder it appears to include PyQt6. Any advice on what/how I need to fix?
Many thanks.

**Comment by roasterdave** [comment]
By "latest version" do you mean the released version 2.8.4 or a continuous build (labeled 2.8.5)?  Does the error occur at start up preventing Artisan from running?  If Artisan does start, what operation causes the error?  Also please post the complete error message.  Thanks.

**Reply by neilpdj** [reply]
Hi, thanks for responding.
Yes, I've been using the released version "artisan-win-2.8.4"
Correct - the error occurs at startup and prevents the app from running.

The error message is:
**Unhandled execption in script
Failed to execute script 'artisan' due to unhandled exception: No module named 'PyQt5'

Traceback (most recent call last):
File "artisanlib\main.py", line 109, in <module>
ImportError: DLL load failed while importing QtWidgets: The specified procedure could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "artisan.py", line 76, in <module>
  File "PyInstaller\loader\pyimod02_importers.py", line 385, in exec_module
  File "artisanlib\main.py", line 135, in <module>
ModuleNotFoundError: No module named 'PyQt5'**

I should also say, this same hardware was running a very old version (1.2 or something) several years back with no problems, if that's at all relevant?!

**Reply by roasterdave** [reply]
Thank you for that additional information.  From the error message it seems that the PyQt6 files were not properly created during the installation.  When any of them are missing Artisan will try to load PyQt5, which is not part of the Windows 10 install package, thus the error.  

Please try this.  Using the Windows file explorer go to C:\Program Files\Artisan and double click uninst.exe to run the uninstaller.  If there are any files or folders left behind in C:\Program Files\Artisan they should be manually deleted, then delete the C:\Program Files\Artisan folder.  Next, run the Artisan v.2.8.4 setup file.  Please let us know if this clears the problem, or not.

**Reply by neilpdj** [reply]
Thanks for taking the time to explain this and walk me through the re-installation process.
Unfortunately, I am experiencing the same result.
Not sure if it helps, but I can see inside the C:\Program Files\Artisan folder that there is a PyQt6 folder and this contains many .pyd files, along with a 'Qt6' folder.

Is there any other way I can install this module?
Thanks again.

**Reply by roasterdave** [reply]
The installer puts the various files where they need to be.  Did you try installing PyQt6?  That is only useful if you run from the source code.  There is no other way to install these components when using the packaged install.  

This could be challenging to diagnose.  Here are two things that will help to understand the state of the install.  

1. Let's compare the contents of the PyQt6 with a functional install.  Open a cmd window and type the following two lines at the command line prompt  
**`cd %temp%`**  
**`dir /Q /S "C:\Program Files\Artisan\PyQt6 > dirlist.txt"`**  


2. Please try installing this [special build](https://www.dropbox.com/scl/fi/nssgezg35pxk9ocatvegf/artisan-win-2.8.5.zip?rlkey=6l25c0511dufebl0ty78is0o4&dl=0).  It is the same code as release v2.8.4, so it will fail as before, but it has the added ability to write a log message when PyQt6 fails to load.  This will hopefully tell us something, but it might not.  There is no need to manually uninstall, just let this installer do the work for you.  


After starting Artisan, type **`%temp%`** into the address bar of the Windows File Explorer and hit Enter.  Select the file **`bootlog.log`** and drag it onto a reply on this web page (do reply with it attached to an email).  Also drag the file **`dirlist.txt`** onto the reply.

Thanks for the information.

**Reply by neilpdj** [reply]
Hi again,
No, I didn't try installing PyQt6 manually.
Attached are the 2 log files. Thank you for taking the time to walk me through this so clearly - it's really appreciated.
Hopefully this helps figure out what's causing the issue.

[bootlog.log](https://github.com/artisan-roaster-scope/artisan/files/12169816/bootlog.log)
[dirlist.txt](https://github.com/artisan-roaster-scope/artisan/files/12169817/dirlist.txt)

**Reply by neilpdj** [reply, marked as answer]
Hi @roasterdave 
Thank you for all your help with this - I found some Windows updates that hadn't been installed. Once Windows was updated, the problem went away. I really appreciate your time. Many thanks,

**Reply by roasterdave** [reply]
@neilpdj Good to hear the install and Artisan are working now.  Thanks for letting us know.

**Comment by MAKOMO** [comment]
Did you install from [artisan-win-2.8.4.zip](https://github.com/artisan-roaster-scope/artisan/releases/download/v2.8.4/artisan-win-2.8.4.zip) which is for Windows 10 or from [artisan-win-legacy-2.8.4.zip](https://github.com/artisan-roaster-scope/artisan/releases/download/v2.8.4/artisan-win-legacy-2.8.4.zip) which is for Windows 8?

**Reply by neilpdj** [reply]
Hi @MAKOMO 
Thanks for replying - I used the "artisan-win-2.8.4.zip" file.
